### PR TITLE
Add getAttribute() to SpanData and EventData

### PR DESCRIPTION
**Pull request by breedx-splk**
Quite a few places where we call `getAttributes().get(key)` on `SpanData` (and spans) and also `EventData`. This adds the singular `getAttribute()` to simplify it a bit and fix a small law of Demeter violation (technicality).

### CODE REVIEW

**Comment by jkwatson** [comment]
Outside of testing use-cases, I'm not sure this is actually going to end up being a very useful addition to the *Data APIs. The main consumer of them is exporters, and exporters will (almost) always want to just grab the full collection of attributes and iterate over them.

I'm not saying this is bad, but it might be more of an internal use-case for tests, that we might be able to tackle via our testing module.

**Comment by breedx-splk** [comment]
> Outside of testing use-cases, I'm not sure this is actually going to end up being a very useful addition to the *Data APIs. The main consumer of them is exporters, and exporters will (almost) always want to just grab the full collection of attributes and iterate over them.
> 
> I'm not saying this is bad, but it might be more of an internal use-case for tests, that we might be able to tackle via our testing module.

Yeah that's fair. 🤷🏻

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java [hunk 1]
```diff
@@ -93,6 +95,16 @@ default String getParentSpanId() {
    */
   Attributes getAttributes();
 
+  /**
+   * Returns the value for a given attribute key.
```
Comment (anuraaga) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java:99]: or null if the attribute is not set

**Review by anuraaga — APPROVED** [review]

**Comment by anuraaga** [comment]
I'm ok with this but it's true that for end users this should generally be covered by the assertj helper

**Comment by jack-berg** [comment]
I think there's a design principle / [best practice](https://github.com/open-telemetry/opentelemetry-java/blob/main/CONTRIBUTING.md#best-practices-that-we-follow) being implicitly discussed with this PR: Do we favor a smaller API surface area or one that's more ergonomic?

In practice we're somewhere in the middle. For example we recently added `AttributesBuilder#removeIf(..)` method despite it being possible, though inconvenient, to take an `Attributes` and remove select entries from it.

I suggest we adopt the practice of limiting the API surface area, unless there stands to be a non-trivial (alas, a subjective decision) performance benefit from extending it.

**Comment by breedx-splk** [comment]
`homer-simpson-quietly-backing-into-the-bushes.gif` 🤪

**Comment by jkwatson** [comment]
> I think there's a design principle / [best practice](https://github.com/open-telemetry/opentelemetry-java/blob/main/CONTRIBUTING.md#best-practices-that-we-follow) being implicitly discussed with this PR: Do we favor a smaller API surface area or one that's more ergonomic?
> 
> In practice we're somewhere in the middle. For example we recently added `AttributesBuilder#removeIf(..)` method despite it being possible, though inconvenient, to take an `Attributes` and remove select entries from it.
> 
> I suggest we adopt the practice of limiting the API surface area, unless there stands to be a non-trivial (alas, a subjective decision) performance benefit from extending it.

I think there's another case that isn't performance-related, and that's developer UX (not us, but our users). In this particular case, I think it wasn't going to help people outside the project too much, even if it would make our lives a tad easier.

**Pull request closed without merging**